figma/lib/fetch_components.ts

```diff
@@ -0,0 +1,246 @@
+  theme?: string
+) {
+  const name = name_.replace(/\s/gi, '-')
+  const id = (theme && themes[theme]?.id) ?? id_
+  return `${process.env.STORYBOOK_FIGMA_URL}/file/${fileId}/${name}?node-id=${encodeURIComponent(id)}`
+}
+
+export type PropType = 'string' | 'boolean' | 'number' | 'object' | 'array' | 'null' | 'JSX.Element'
+export type PropTypes = Array<PropType>
+export type PropData = { required: boolean; name: string; values?: string[]; type?: PropTypes; items?: PropData }
+export type Props = Record<string, PropData>
+type RawComponentsType = {
+  variants: Record<string, Pick<Node<'COMPONENT'>, 'name'>>
+  theme?: string
+}
+export type ThemeData = Record<string, { id: string; theme: string }>
+type MergedComponentType = Omit<RawComponentsType, 'theme'> & { themes?: ThemeData }
+
+type ComponentsType = { states: string[]; props: Props; themes?: ThemeData }
+type CommonComponentInfo = Pick<Node<'COMPONENT_SET'>, 'name' | 'id'> & {
+  path: string
+  canvasName: string
+  frameId: string
+  frameName: string
+}
+type RawOutput<Type extends {} = RawComponentsType> = Record<string, RawComponentInfo<Type>>
+type RawComponentInfo<T extends {}> = CommonComponentInfo & T
+
+export type FileInfo = { name: string; id: string }
+
+export type Output = RawOutput<ComponentsType>
+export type ComponentInfo = RawComponentInfo<ComponentsType>
+
+const STATE_VALUES_DICTIONARY = {
+  focused: 'focus',
+  pressed: 'active',
+  disable: 'disabled',
+  disabled: 'disabled',
+  readOnly: 'readonly',
+}
+function cleanStateValues(states: string[]) {
+  return states.map((v) => STATE_VALUES_DICTIONARY[v] ?? v)
+}
+
+function cleanOutput<T extends RawOutput>(output: T): T {
+  const res: T = {} as T
+  Object.keys(output).forEach((key: Extract<string, keyof T>) => {
+    const comp = output[key]
+    if (key.match(/Group\s[0-9]+/g)) return
+    res[key] = comp
+  })
+  return res
+}
+
+function combineThemes(output: RawOutput): RawOutput<MergedComponentType> {
+  let compKeys = Object.keys(output)
+  const components = compKeys.reduce((agg, compKey) => {
+    const cmp = output[compKey]
+    const theme = cmp.theme ?? 'global'
+    if (agg[cmp.path]) {
+      agg[cmp.path].variants = { ...agg[cmp.path].variants, ...cmp.variants }
+    } else {
+      agg[cmp.path] = cmp
+      agg[cmp.path].themes = {}
+    }
+    agg[cmp.path].themes[theme] = { id: cmp.id, theme }
+    delete cmp.theme
+    return agg
+  }, {} as RawOutput<MergedComponentType>)
+  compKeys = Object.keys(components)
+  compKeys.forEach((compKey) => {
+    const cmp = components[compKey]
+    if ((Object.keys(cmp.themes)?.length ?? 0) < 2) delete cmp.themes
+  })
+  return components
+}
+
+function variantNameToKeyValues(variantName: string) {
+  return variantName
+    .split(',')
+    .filter(Boolean)
+    .map((keyValue) =>
+      keyValue
+        .trim()
+        .split('=')
+        .reduce((_, __, ___, [key, value]) => ({ key: camelize(key.trim()), value: camelize((value ?? '').trim()) }), {
```

Comment:
In general `eslint` will scream if a variable is unused (memory!), and try to remove them - if it can, but sometimes you need to have unused variables due to, like in this instance, the order of function arguments, or perhaps in an object spread you would like to skip a variable:
```ts
const { notNeeded: _, ....weCareAboutTheRest } = whatever 
```
Basically underscore in the beginning means that we do not care about it at all, but as we have to skip the whole 3 arguments, you end up with `_, __, __,` (it could have been `_a, _b, _c` as well btw)

Reply:
I don't think uninterested arguments should be named, because, you literally reject them. It is a common pattern AFAIK. As far as the code is concerned it's meant to be all build on lambdas, I like functional programming, but if this wasn't clear I'll change it